server/src/handlers/event.rs

```diff
@@ -74,7 +80,16 @@ pub async fn query(_req: HttpRequest, json: web::Json<Value>) -> HttpResponse {
 
 pub async fn post_event(req: HttpRequest, body: web::Json<serde_json::Value>) -> HttpResponse {
     let stream_name: String = req.match_info().get("logstream").unwrap().parse().unwrap();
-    let labels = utils::collect_labels(&req);
+
+    let tags = match collect_labelled(&req, PREFIX_TAGS, ',') {
+        Ok(tags) => HashMap::from([("p_tags".to_string(), tags)]),
+        Err(e) => return e.error_response(),
+    };
+
+    let metadata = match collect_labelled(&req, PREFIX_META, ',') {
```

Comment:
Lets have a multi levelled approach here --

`collect_tags` and `collect_meta` as top level functions, both calling  private fn `collect_from_header` with proper prefix and request.